src/combineReducers.js

```diff
@@ -90,50 +132,18 @@ function assertReducerSanity(reducers) {
  * @returns {Function} A reducer function that invokes every reducer inside the
  * passed object, and builds a state object with the same shape.
  */
-export default function combineReducers(reducers) {
-  var reducerKeys = Object.keys(reducers)
-  var finalReducers = {}
-  for (var i = 0; i < reducerKeys.length; i++) {
-    var key = reducerKeys[i]
-    if (typeof reducers[key] === 'function') {
-      finalReducers[key] = reducers[key]
-    }
-  }
-  var finalReducerKeys = Object.keys(finalReducers)
 
-  var sanityError
-  try {
-    assertReducerSanity(finalReducers)
-  } catch (e) {
-    sanityError = e
-  }
+export default function combineReducers(config) {
+  var reducers = getReducers(config)
+  var keys = Object.keys(reducers)
 
   return function combination(state = {}, action) {
-    if (sanityError) {
-      throw sanityError
-    }
+    assertReducerSanity(getReducers(config))
```

Comment:
This is expensive. This calculates `getReducers` on every dispatch but they never change.